### PR TITLE
osd: minor performance improvements and fixes

### PR DESCRIPTION
**Pull request by xiexingguo**

### CODE REVIEW

**Comment by xiexingguo** [comment]
@athanatos Please take a look. Thanks!

File: src/osd/OSD.h [hunk 1]
```diff
@@ -949,13 +949,13 @@ class OSDService {
     Mutex::Locker l(recovery_lock);
     _maybe_queue_recovery();
   }
-  void clear_queued_recovery(PG *pg, bool front = false) {
+  void clear_queued_recovery(PG *pg) {
```
Comment (athanatos) [inline src/osd/OSD.h]: Actually, this one may come to life again soon, please remove this commit.

File: src/common/hobject.h [hunk 2]
```diff
@@ -250,7 +250,8 @@ struct hobject_t {
   }
   void set_bitwise_key_u32(uint32_t value) {
     hash = _reverse_bits(value);
-    build_hash_cache();
```
Comment (athanatos) [inline src/common/hobject.h]: Please add a comment to this and build_hash_cache() to ensure that we remember to update both.

**Comment by xiexingguo** [comment]
@athanatos Done and thanks for the confirmation.

**Comment by athanatos** [comment]
@xiexingguo I haven't reviewed those 4 new ones.  I'd appreciate it if you did not add more commits to a PR after I've reviewed the PR and added the needs-qa label...  Please remove them and ping me so that I can re-add the needs-qa label.

**Comment by xiexingguo** [comment]
> I'd appreciate it if you did not add more commits to a PR after I've reviewed the PR and added the needs-qa label...

@athanatos Sure.
Moved those changes into https://github.com/ceph/ceph/pull/10548, and please take a look at them too. Thanks!

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2016-08-05_14:30:29-rados-wip-yuri-testing_2016_08_05-distro-basic-smithi/

**Pull request merged**